### PR TITLE
Add /weekly and /monthly to Telegram (#5527)

### LINKED ISSUE #5527: Show average profit in overview
## Describe your environment
(if applicable)

  * Operating system: ____
  * Python Version: _____ (`python -V`)
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  * Freqtrade Version: ____ (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
## Describe the enhancement

I would like to see (daily / monthly / yearly) average profit in the /profit RPC 

Example: 

ROI: Closed trades
∙ -
∙ -
ROI: All trades
∙ -
∙ -

Total Trade Count: -
First Trade opened: -
Latest Trade opened: 15 hours ago
Win / Loss: 68 / 12
Average Weekly Profit: -% (100 USDT)
Avg. Duration: TIME
Best Performing: PAIR

*Explain the enhancement you would like*

### PR DESCRIPTION
**Pull request by Merinorus**
## Summary

Add /weekly and /monthly commands to Telegram RPC

Closes #5527 show average profit in overwiew (more or less)

## Quick changelog

- /weekly <n> shows profit & loss the last n ISO weeks (starting from Monday)
- /monthly <n> shows profit & loss the last n months

## What's new?

```
/weekly

Weekly Profit over the last 8 weeks (starting from Monday):
Monday      Profit BUSD    Profit EUR    Trades
----------  -------------  ------------  --------
2021-11-01  0.000 BUSD     0.000 EUR     0 trades
2021-10-25  0.000 BUSD     0.000 EUR     0 trades
2021-10-18  0.000 BUSD     0.000 EUR     0 trades
2021-10-11  0.000 BUSD     0.000 EUR     0 trades
2021-10-04  0.000 BUSD     0.000 EUR     0 trades
2021-09-27  0.000 BUSD     0.000 EUR     0 trades
2021-09-20  0.000 BUSD     0.000 EUR     0 trades
2021-09-13  0.000 BUSD     0.000 EUR     0 trades


/monthly

Monthly Profit over the last 6 months:
Month    Profit BUSD    Profit EUR    Trades
-------  -------------  ------------  --------
2021-11  0.000 BUSD     0.000 EUR     0 trades
2021-10  0.000 BUSD     0.000 EUR     0 trades
2021-09  0.000 BUSD     0.000 EUR     0 trades
2021-08  26.109 BUSD    22.621 EUR    1 trades
2021-07  -0.591 BUSD    -0.512 EUR    1 trades
2021-06  0.000 BUSD     0.000 EUR     0 trades
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -42,3 +42,6 @@ colorama==0.4.4
 # Building config files interactively
 questionary==1.10.0
 prompt-toolkit==3.0.21
+# Extensions to datetime library
+types-python-dateutil==2.8.2
```
Comment (xmatthias) [inline requirements.txt]: I think the `types` dependency should go into `requirements-dev.txt` ... it's not necessary on an installation that ain't running `mypy`.

File: freqtrade/rpc/rpc.py [hunk 2]
```diff
@@ -289,6 +290,91 @@ def _rpc_daily_profit(
             'data': data
         }
 
+    def _rpc_weekly_profit(
+            self, timescale: int,
+            stake_currency: str, fiat_display_currency: str) -> Dict[str, Any]:
+        today = datetime.utcnow().date()
```
Comment (xmatthias) [inline freqtrade/rpc/rpc.py]: we should try to not use `utcnow()` - but instead use `now(timezone.utc)`.

utcnow() will return a timezone native object - processing this further can result in the wrong day depending on the timezone offset you're in (especially near midnight).

**Review by xmatthias — COMMENTED** [review]
Great PR already.
I've only had a quick look (will have to try this later) ... but it looks already pretty good!

File: freqtrade/rpc/rpc.py (hunk 2 above)
Comment (Merinorus) [reply freqtrade/rpc/rpc.py]: Ok got it. The /daily commands uses also the timezone naive method: `today = datetime.utcnow().date()`
Should we change it now or it should be included in a bugfix PR?

Comment (xmatthias) [reply freqtrade/rpc/rpc.py]: good catch - i'd leave that up to you ... :)

Comment (Merinorus) [reply freqtrade/rpc/rpc.py]: Ok great! It will be in another commit in case it needs to be reverted.

**Comment by Merinorus** [comment]
All good for me!

**Review by xmatthias — APPROVED** [review]
> All good for me!

For me too :+1: 

this is a great new feature - looking forward to using it already!

**Pull request merged**